prefect_gcp/workers/cloud_run_v2.py

```diff
@@ -125,6 +131,7 @@ class CloudRunWorkerJobV2Configuration(BaseJobConfiguration):
             "complete before raising an exception."
         ),
     )
+    job_name: str = Field(default=None, description="The name of the Cloud Run job.")
```

Comment:
When we move `job_name` from a property to a Pydantic field, it becomes modifiable within the `base_job_template`. We'll either need to take that into account in `_populate_job_name` or keep `job_name` as a property. I lean towards keeping `job_name` a property since a user would run into the errors this PR fixes if they set `job_name` in the base job template.